app/services/pda/provider_details.rb

```diff
@@ -1,30 +1,38 @@
 module PDA
   class ProviderDetails
     ApiError = Class.new(StandardError)
+    ValidDetailsNotFound = Class.new(StandardError)
 
     # Only save schedule details that are relevant to civily apply
     APPLICABLE_CATEGORIES_OF_LAW = %w[MAT].freeze
     APPLICABLE_AREAS_OF_LAW = ["LEGAL HELP", "CIVIL FUNDING"].freeze
 
-    def initialize(office_code)
+    def initialize(provider, office_code)
+      @provider = provider
       @office_code = office_code
     end
 
-    def self.call(office_code)
-      new(office_code).call
+    def self.call(provider, office_code)
+      new(provider, office_code).call
     end
 
     def call
       raise ApiError, "API Call Failed: (#{response.status}) #{response.body}" unless response.success?
 
       destroy_existing_schedules
+
+      # TODO??: create/update firm and office if they exists in PDA at all rather than if exists WITH a schedule??
+      # Since schedules can expire we would eventually have offices without active schedules anyway so why not just create
+      # and update the firm/office regardless, then handle schedule logic
       if response.status == 200
         update_firm
         update_office
+        update_provider
         create_schedules
         Rails.logger.info("#{self.class} - No applicable schedules found for #{@office_code}") if office.schedules.empty?
       else
         Rails.logger.info("#{self.class} - No schedules found for #{@office_code}")
+        raise ValidDetailsNotFound, "No valid details found for office account number #{@office_code}"
```

Comment:
Just a question from me, do we want to raise an error everytime PDA doesn't find schedule details for a provider? I am thinking sentry/alert channels might become quite noisy. I wasn't sure whether this was really an error scenario, or just something we should be trying to handle gracefully, hopefully by redirecting them to an appropriate page/preventing them from progressing any further.

Reply:
this specific error is caught and handled so will not be captured by sentry in any event. We may well change the mechanism in the future too. it was just one obvious way to handle.